scripts/route_check.py

```diff
@@ -0,0 +1,195 @@
+    return mode
+
+def print_message(lvl, *args):
+    if (lvl <= mode):
+        for arg in args:
+            print arg
+
+def add_prefix(ip):
+    if ip.find(IPV6_SEPARATOR) == -1:
+        ip = ip + PREFIX_SEPARATOR + "32"
+    else:
+        ip = ip + PREFIX_SEPARATOR + "128"
+    return ip
+
+def add_prefix_ifnot(ip):
+    return ip if ip.find(PREFIX_SEPARATOR) != -1 else add_prefix(ip)
+
+def ip_subnet(ip):
+    if ip.find(":") == -1:
+        net = ipaddress.IPv4Network(ip, False)
+    else:
+        net = ipaddress.IPv6Network(ip, False)
+    return net.with_prefixlen
+
+def cmps(s1, s2):
+    if (s1 == s2):
+        return 0
+    if (s1 < s2):
+        return -1
+    return 1
+
+def do_diff(t1, t2):
+    t1_x = t2_x = 0
+    t1_miss = []
+    t2_miss = []
+    t1_len = len(t1);
+    t2_len = len(t2);
+    while t1_x < t1_len and t2_x < t2_len:
+        d = cmps(t1[t1_x], t2[t2_x])
+        if (d == 0):
+            t1_x += 1
+            t2_x += 1
+        elif (d < 0):
+            t1_miss.append(t1[t1_x])
+            t1_x += 1
+        else:
+            t2_miss.append(t2[t2_x])
+            t2_x += 1
+
+    while t1_x < t1_len:
+        t1_miss.append(t1[t1_x])
+        t1_x += 1
+
+    while t2_x < t2_len:
+        t2_miss.append(t2[t2_x])
+        t2_x += 1
+    return t1_miss, t2_miss
+
+
+def get_routes():
+    db = ConfigDBConnector()
+    db.db_connect('APPL_DB')
+    print_message(MODE_DEBUG, "APPL DB connected for routes")
+    keys = db.get_keys('ROUTE_TABLE')
+    print_message(MODE_DEBUG, json.dumps({"ROUTE_TABLE": keys}, indent=4))
+
+    valid_rt = []
+    skip_rt = []
+    for k in keys:
+        if db.get_entry('ROUTE_TABLE', k)['nexthop'] != '':
+            valid_rt.append(add_prefix_ifnot(k))
+        else:
+            skip_rt.append(k)
+
+    print_message(MODE_INFO, json.dumps({"skipped_routes" : skip_rt}, indent=4))
+    return sorted(valid_rt)
+
+def get_route_entries():
+    db = ConfigDBConnector()
+    db.db_connect('ASIC_DB')
+    print_message(MODE_DEBUG, "ASIC DB connected")
+    keys = db.get_keys('ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY', False)
+    print_message(MODE_DEBUG, json.dumps({"ASIC_ROUTE_ENTRY": keys}, indent=4))
+
+    rt = []
+    for k in keys:
+        rt.append(k.split("\"", -1)[3])
+    return sorted(rt)
+
+
+def get_interfaces():
+    db = ConfigDBConnector()
+    db.db_connect('APPL_DB')
+    print_message(MODE_DEBUG, "APPL DB connected for interfaces")
+
+    intf = []
+    keys = db.get_keys('INTF_TABLE')
+    print_message(MODE_DEBUG, json.dumps({"APPL_DB_INTF": keys}, indent=4))
+
+    for k in keys:
+        subk = k.split(':', -1)
+        alias = subk[0]
+        ip_prefix = ":".join(subk[1:])
+        ip = add_prefix(ip_prefix.split("/", -1)[0])
+        if (subk[0] == "eth0") or (subk[0] == "docker0"):
+            continue
+        if (subk[0] != "lo"):
+            intf.append(ip_subnet(ip_prefix))
+        intf.append(ip)
+    return sorted(intf)
+
+def checkout():
```

Comment:
Suggest a more descriptive name for this function. Maybe `check_routes()` or `do_route_check()`?